Tests/js/views/services/plugins/assets/ez-locationsearchplugin-tests.js

```diff
@@ -231,14 +252,35 @@ YUI.add('ez-locationsearchplugin-tests', function (Y) {
                 };
             };
 
+            this.ContentModelConstructor.prototype.loadFromHash = function (hash) {
+                this.hash = hash;
+
+                this.get = function (attr) {
+                    switch (attr) {
+                        case 'id':
+                            return that.contentInfo.id;
+                        default:
+                            Assert.fail('Requested attribute does not exist in the content model');
+                            break;
+                    }
+                };
+            };
             Mock.expect(this.capi, {
                 method: 'getContentService',
                 returns: this.contentService,
             });
             Mock.expect(this.contentService, {
                 method: 'newViewCreateStruct',
-                args: [this.viewName, 'LocationQuery'],
-                returns: this.query,
+                args: [Mock.Value.String, Mock.Value.String],
+                run: function (viewName, queryType) {
+                    if (viewName == that.viewName && queryType == "LocationQuery") {
```

Comment:
I would just use the `queryType` to return the location query or the content query. The view name are more implementation details here.